core/driver/sources/src/main/scala/org/virtuslab/ideprobe/reporting/ErrorValidator.scala

```diff
@@ -18,8 +20,16 @@ object ErrorValidator {
     }
   }
 
+  private def ideMessageMatchesMessagesFromConfig(ideMessage: IdeMessage, messagesFromConfig: Seq[String]): Boolean =
+    messagesFromConfig.exists(configMessage =>
+      trimEachLine(ideMessage.content).contains(trimEachLine(configMessage)) ||
+        configMessage.r.findFirstIn(ideMessage.content).nonEmpty
```

Comment:
Okay... so the match is either based on regex matching a single line, or non-regex literal string matching multiple lines? 🤔 

TBH it's getting pretty tricky/convoluted as for config key... I'd consider `exclude/includeMessages` as they are (regex, single-line) and adding a new pair of keys for matching stack trace specifically (non-regex, multi-line)

Reply:
> Okay... so the match is either based on regex matching a single line, or non-regex literal string matching multiple lines?

Well, not exactly. It will work as described below, both for `probe.driver.check.errors.includeMessages` and `probe.driver.check.errors.excludeMessages`:

1) If user adds a **regex-string** in the config file - it will work well and match error message **as long as the regex-string is NOT multiline**
2) If user adds a **string literal** (without regex usage) - it will always work both for single-line and multiline **string literal**

It might look complicated in the code, but from the user's perspective - if we add a short note in the `reference.conf` - that usage of regular expressions is limited to single-line strings - it should be easy to understand

. I would suggest to keep two config keys: `probe.driver.check.errors.includeMessages` and `probe.driver.check.errors.excludeMessages` and not add new ones. In my opinion adding more keys is the thing that would make configuration more complicated.

Reply:
And such descriptions will be added to `reference.conf` soon - in the https://github.com/VirtusLab/ide-probe/issues/262 issue

Reply:
Okay! makes sense indeed now. I can see some exotic edge cases with false-positives (like, the given string doesn't match when interpreted verbatim as intended, but accidentally matches as a regex)... but I wouldn't care too much

Reply:
Esp. that regex-special characters like `+*?{}` aren't likely to occur in stack traces; only `.` and `()` are prevalent

Reply:
Although.... one thing! what if `configMessage` is an **invalid regex**, like `lol (xd`? won't it end up in an exception on `.r` call?

Reply:
Pls add such a case to the tests, even if it doesn't end up in an exception in the current implementation

Reply:
tested with a `"at com.intellij.psi.impl.file.impl.FileManagerImpl.getCachedPsiFile(F"` string...
```
scala> "at com.intellij.psi.impl.file.impl.FileManagerImpl.getCachedPsiFile(F".r
java.util.regex.PatternSyntaxException: Unclosed group near index 69
at com.intellij.psi.impl.file.impl.FileManagerImpl.getCachedPsiFile(F
  at java.base/java.util.regex.Pattern.error(Pattern.java:2028)
  at java.base/java.util.regex.Pattern.accept(Pattern.java:1878)
  at java.base/java.util.regex.Pattern.group0(Pattern.java:3053)
  at java.base/java.util.regex.Pattern.sequence(Pattern.java:2124)
  at java.base/java.util.regex.Pattern.expr(Pattern.java:2069)
  at java.base/java.util.regex.Pattern.compile(Pattern.java:1783)
  ... 63 elided
```
It will throw an exception. But maybe such exception would be a good thing - so that user will know that the configuration is invalid? `ide-probe` is used for tests so throwing such Exception might not be a bad thing? (it would be bad if `ide-probe` was for runtime...)

Reply:
And yet, before my changes - we had the same possibility for exceptions here - we were just calling `.r` on each string from these configs

Reply:
Right! makes sense indeed, let's leave as is. Up to you whether a test case is needed (maybe not)

Reply:
On the other hand, such scenario seems impossible to me. Since || short circuits and 
`trimEachLine(ideMessage.content).contains(trimEachLine(configMessage))`
would evaluate to `true`... Exception would not be thrown.

Reply:
I mean, in case where the first operand `||` evaluates to false, it's still possible that `.r` is called, right?